lib/plaindate.ts

```diff
@@ -248,6 +248,7 @@ export class PlainDate implements Temporal.PlainDate {
       nanosecond = 0;
     if (temporalTime !== undefined) {
       temporalTime = ES.ToTemporalTime(temporalTime);
+      ES.assertIsPlainTime(temporalTime, 'ToTemporalTime above always returns a PlainTime');
```

Comment:
Will this justification string end up adding to bundle size? If so, then maybe make this a comment?

If we really wanted to be fancy, we could create a custom lint rule that verified that any use of `ES.assert*` functions was preceded by a comment with a specific format, e.g. "// justification: ".  If you've never written a custom lint rule before, it's surprisingly easy.

Reply:
> Will this justification string end up adding to bundle size? If so, then maybe make this a comment?

No - terser removes the function and calls entirely because to Terser/JS it looks like an empty function with all it's args unused.

Reply:
Cool. One more thought: having the justification string as an argument might cause naive readers to think that the function actually does something with that string, e.g. some sort of runtime validation. I wonder if we should name the function differently (e.g. `changeType` or `noOpAssert` or something like that) and/or remove the justification arg to clarify this?

Reply:
I renamed this to `uncheckedAssertNarrowedType` as that is what I intend for this function to allow:
 - it's not being checked at runtime
 - ideally, it should allow "narrowing" down a union of types to one of the union's members. I've not quite found a way to do this in the type system, but I'll continue to iterate on this with the help of some teammates over the coming weeks.